lib/connection/msg.js

```diff
@@ -93,26 +93,14 @@ class Msg {
     buffers.push(header);
 
     let totalLength = header.length;
-
     const command = this.command;
-    const nameArgumentPair = validSegmentListNamePairs(command);
-    if (nameArgumentPair) {
-      // TODO: Add support for payload type 1
-      const argument = nameArgumentPair.argument;
-
-      // Add initial type 0 segment with arguments pulled up
-      const clonedQuery = Object.assign({}, command);
-      delete clonedQuery[argument];
-      totalLength += this.makeDocumentSegment(buffers, clonedQuery);
-
-      // Create type 1 query
-      totalLength += this.makeSequenceSegment(buffers, argument, command[argument]);
-    } else {
-      totalLength += this.makeDocumentSegment(buffers, command);
-    }
-
-    writeInt32ListToUint8Buffer(header, [totalLength, this.requestId, 0, opcodes.OP_MSG, flags]);
+    totalLength += this.makeDocumentSegment(buffers, command);
 
+    header.writeInt32LE(totalLength, 0); // messageLength
```

Comment:
Can any of these values be negative? I think to guard ourselves we should make the `writeUInt32LE`?

Reply:
The spec itself defines them as `int32`. Though I agree I can't imagine why they would be negative I wouldn't want to be non-compliant if they ever decide to do something with negative values here.